kube-proxy panics with SIGSEGV on first run

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:
When creating a new `1.18.0` cluster using `kubeadm`, `kube-proxy` panics with SIGSEGV on first run:

```
I0401 11:10:20.858107       1 node.go:136] Successfully retrieved node IP: [redacted]
I0401 11:10:20.858228       1 server_others.go:259] Using ipvs Proxier.
W0401 11:10:20.859035       1 proxier.go:429] IPVS scheduler not specified, use rr by default
I0401 11:10:20.859307       1 server.go:583] Version: v1.18.0
I0401 11:10:20.860132       1 conntrack.go:100] Set sysctl 'net/netfilter/nf_conntrack_max' to 131072
I0401 11:10:20.860212       1 conntrack.go:52] Setting nf_conntrack_max to 131072
I0401 11:10:20.860437       1 conntrack.go:100] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_established' to 86400
I0401 11:10:20.860540       1 conntrack.go:100] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_close_wait' to 3600
I0401 11:10:20.860862       1 config.go:133] Starting endpoints config controller
I0401 11:10:20.860878       1 config.go:315] Starting service config controller
I0401 11:10:20.860890       1 shared_informer.go:223] Waiting for caches to sync for service config
I0401 11:10:20.860893       1 shared_informer.go:223] Waiting for caches to sync for endpoints config
I0401 11:10:20.961249       1 shared_informer.go:230] Caches are synced for service config 
I0401 11:10:20.961244       1 shared_informer.go:230] Caches are synced for endpoints config 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x1514fb8]

goroutine 81 [running]:
k8s.io/kubernetes/pkg/proxy/ipvs.(*Proxier).syncService(0xc00000a3c0, 0xc000413800, 0x18, 0xc0001f8280, 0xc000243101, 0x5, 0xc00053a9ad)
        /workspace/anago-v1.18.0-rc.1.21+8be33caaf953ac/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/proxy/ipvs/proxier.go:1903 +0x48
k8s.io/kubernetes/pkg/proxy/ipvs.(*Proxier).syncProxyRules(0xc00000a3c0)
        /workspace/anago-v1.18.0-rc.1.21+8be33caaf953ac/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/proxy/ipvs/proxier.go:1186 +0xcdd
k8s.io/kubernetes/pkg/proxy/ipvs.(*Proxier).OnEndpointsSynced(0xc00000a3c0)
        /workspace/anago-v1.18.0-rc.1.21+8be33caaf953ac/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/proxy/ipvs/proxier.go:890 +0x62
k8s.io/kubernetes/pkg/proxy/config.(*EndpointsConfig).Run(0xc000796100, 0xc00009a0c0)
        /workspace/anago-v1.18.0-rc.1.21+8be33caaf953ac/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:141 +0x108
created by k8s.io/kubernetes/cmd/kube-proxy/app.(*ProxyServer).Run
        /workspace/anago-v1.18.0-rc.1.21+8be33caaf953ac/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:701 +0xb7f
```

Note that after the crash the pod restarts, and the second run is successful:

```
I0401 11:10:22.387131       1 node.go:136] Successfully retrieved node IP: [redacted]
I0401 11:10:22.387187       1 server_others.go:259] Using ipvs Proxier.
W0401 11:10:22.390681       1 proxier.go:429] IPVS scheduler not specified, use rr by default
I0401 11:10:22.390993       1 server.go:583] Version: v1.18.0
I0401 11:10:22.392209       1 conntrack.go:52] Setting nf_conntrack_max to 131072
I0401 11:10:22.392555       1 config.go:133] Starting endpoints config controller
I0401 11:10:22.392574       1 shared_informer.go:223] Waiting for caches to sync for endpoints config
I0401 11:10:22.392588       1 config.go:315] Starting service config controller
I0401 11:10:22.392592       1 shared_informer.go:223] Waiting for caches to sync for service config
I0401 11:10:22.492824       1 shared_informer.go:230] Caches are synced for service config 
I0401 11:10:22.492976       1 shared_informer.go:230] Caches are synced for endpoints config
```

and everything seems to be working as expected.

**What you expected to happen**:

`kube-proxy` starts successfully the first time.

**How to reproduce it (as minimally and precisely as possible)**:

Create a new `1.18.0` cluster using `kubeadm`, with `kube-proxy` in `ipvs` mode.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): **1.18.0**
- Cloud provider or hardware configuration: **Hetzner VPS**
- OS (e.g: `cat /etc/os-release`): **Ubuntu 18.04.4 LTS**
- Kernel (e.g. `uname -a`): **5.3.0-45-generic** (Ubuntu's default HWE kernel)
- Install tools: **kubeadm**
- Network plugin and version (if this is a network-related bug):
- Others: using `cri-o 1.17.2` CRI and `kube-proxy` `ipvs` mode

Please let me know if you need any additional information.

--- a/pkg/proxy/ipvs/proxier.go
+++ b/pkg/proxy/ipvs/proxier.go
@@ -608,7 +608,7 @@ func (handle *LinuxKernelHandler) GetModules() ([]string, error) {
 	}
 	ipvsModules := utilipvs.GetRequiredIPVSModules(kernelVersion)
 
-	var bmods []string
+	var bmods, lmods []string
 
 	// Find out loaded kernel modules. If this is a full static kernel it will try to verify if the module is compiled using /boot/config-KERNELVERSION
 	modulesFile, err := os.Open("/proc/modules")
@@ -650,11 +650,15 @@ func (handle *LinuxKernelHandler) GetModules() ([]string, error) {
 			if err != nil {
 				klog.Warningf("Failed to load kernel module %v with modprobe. "+
 					"You can ignore this message when kube-proxy is running inside container without mounting /lib/modules", module)
+			} else {
+				lmods = append(lmods, module)
 			}
 		}
 	}
 
-	return append(mods, bmods...), nil
+	mods = append(mods, bmods...)
+	mods = append(mods, lmods...)
+	return mods, nil
 }
 
 // getFirstColumn reads all the content from r into memory and return a

